### PR TITLE
[WIP] xgblauncher setup

### PR DESCRIPTION
**Pull request by sperlingxx**

### CODE REVIEW

**Comment by sperlingxx** [comment]
@yinlou @jq  @terrytangyuan Can you help to review this PR?

**Comment by ywskycn** [comment]
> @yinlou @jq @terrytangyuan Can you help to review this PR?

Better to split into several smaller PRs...

File: xgboost-launcher/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        pass
+```
+
+#### install XGBoost-Launcher
+
+XGBoost-Launcher is a python package, which should install separately.
+
+Due to auto-training integration, for now, XGBoost-Launcher is only compatible with Ant-XGBoost.
+
+By the way, XGBoost-Launcher requires python >= 3.6.
```
Comment (terrytangyuan) [inline xgboost-launcher/README.md:57]: Could you go through and fix all the grammar issues this document might contain? Also make this README more formal, e.g. "by the way" should not appear here.

File: xgboost-launcher/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    @abstractmethod
+    def write_lines(self, lines: List[str], model_path: str):
+        pass
+```
+
+#### install XGBoost-Launcher
+
+XGBoost-Launcher is a python package, which should install separately.
+
+Due to auto-training integration, for now, XGBoost-Launcher is only compatible with Ant-XGBoost.
```
Comment (terrytangyuan) [inline xgboost-launcher/README.md:55]: What is auto-training integration? Provide some reference here.

File: xgboost-launcher/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+By the way, XGBoost-Launcher requires python >= 3.6.
+
+Running below codes, XGBoost-Launcher and Ant-XGBoost it depended on will be installed.
```
Comment (terrytangyuan) [inline xgboost-launcher/README.md:59]: What is "it" here?

File: xgboost-launcher/launcher/data_source.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        self._sources[name] = config_def
+        self._source_impls[name] = source_impl
+        logger.info("Data source %s registered successfully!" % name)
+
+    def create_init_fn(self, name, config):
+        assert name in self._sources, "Unregistered data source %s!" % name
+        assert isinstance(config, dict)
+        # convert Dict to target data source config (NamedTuple)
+        data_source_conf = self._sources[name](**config)
+        return lambda rank, world_size, col_format: \
```
Comment (terrytangyuan) [inline xgboost-launcher/launcher/data_source.py:74]: avoid the use of `\`

File: xgboost-launcher/launcher/data_source.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            self._source_impls[name](rank, world_size, col_format, data_source_conf)
+
+
+__DataSources = DataSourceCollection()
```
Comment (terrytangyuan) [inline xgboost-launcher/launcher/data_source.py:78]: Remove redundant underscore

**Review by terrytangyuan — COMMENTED** [review]
Agree with @ywskycn. This PR has 3000+ lines of change which make it impossible to review.

**Pull request closed without merging**